controllers/scopeinstance_controller.go

```diff
@@ -77,11 +82,37 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	log.Log.Info("Reconciling ScopeInstance", "namespaceName", req.NamespacedName)
 
-	in := &operatorsv1.ScopeInstance{}
-	if err := r.Client.Get(ctx, req.NamespacedName, in); err != nil {
-		return ctrl.Result{}, err
+	existingIn := &operatorsv1.ScopeInstance{}
+	if err := r.Client.Get(ctx, req.NamespacedName, existingIn); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// Perform reconciliation
+	reconciledIn := existingIn.DeepCopy()
+	res, reconcileErr := r.reconcile(ctx, reconciledIn)
+
+	// Update the status subresource before updating the main object. This is
+	// necessary because, in many cases, the main object update will remove the
+	// finalizer, which will cause the core Kubernetes deletion logic to
+	// complete. Therefore, we need to make the status update prior to the main
+	// object update to ensure that the status update can be processed before
+	// a potential deletion.
+	if !equality.Semantic.DeepEqual(existingIn.Status, reconciledIn.Status) {
+		if updateErr := r.Client.Status().Update(ctx, reconciledIn); updateErr != nil {
+			return res, apimacherrors.NewAggregate([]error{reconcileErr, updateErr})
+		}
+	}
+	existingIn.Status, reconciledIn.Status = operatorsv1.ScopeInstanceStatus{}, operatorsv1.ScopeInstanceStatus{}
+	if !equality.Semantic.DeepEqual(existingIn, reconciledIn) {
+		if updateErr := r.Client.Update(ctx, reconciledIn); updateErr != nil {
+			return res, apimacherrors.NewAggregate([]error{reconcileErr, updateErr})
+		}
+	}
```

Comment:
I am not sure if this section is actually necessary as I don't think we ever mess with anything on the CRs other than the status. Would we be okay to remove this?